static/js/services/delete-docs.js

```diff
@@ -75,22 +108,33 @@ var _ = require('underscore');
         docs.forEach(function(doc) {
           doc._deleted = true;
         });
+        var parents = [];
         return $q.all(docs.map(function(doc) {
           return getParent(doc)
             .then(function(parent) {
               if (parent) {
-                docs.push(parent);
+                parents.push(parent);
               }
             });
           }))
           .then(function() {
-            return checkForDuplicates(docs);
-          })
-          .then(function() {
-            return minifyLineage(docs);
-          })
-          .then(function() {
-            return DB().bulkDocs(docs);
+            var docsWithParents = docs.concat(parents);
+            checkForDuplicates(docsWithParents);
+
+            if (!Session.isAdmin()) {
+              minifyLineage(docsWithParents);
+              return DB().bulkDocs(docsWithParents);
+            }
+
+            var docIds = docs.map(function(doc) {
+              return { _id: doc._id };
+            });
+            return $q.all([
+              parents.length > 0 ? DB().bulkDocs(parents) : $q.resolve([]),
```

Comment:
I don't get why this is happening.

Reply:
So for offline users, the docs get deleted and parents get updated in one `DB().bulkDocs` call, but for online users the docs get deleted in one call, and the parents get updated in another. A lot of the time parents won't need to be updated as the hierarchy is fine even with the deletions, so I added this check in here to avoid an unnecessary `DB().bulkDocs` call. This could probably be made clearer with some more sensible code branching though